neuraxle/pipeline.py

```diff
@@ -71,7 +60,27 @@ class Pipeline(BasePipeline, ResumableStepMixin):
     """
 
     def __init__(self, steps: NamedTupleList):
-        super().__init__(steps)
+        BasePipeline.__init__(self, steps=steps)
+
+    def transform(self, data_inputs: Any):
+        """
+        After loading the last checkpoint, transform each pipeline steps
+        :param data_inputs: the data input to transform
+        :return: transformed data inputs
+        """
+        self.setup()
+
+        current_ids = self.hasher.hash(
+            current_ids=None,
+            hyperparameters=self.hyperparams,
+            data_inputs=data_inputs
+        )
+        data_container = DataContainer(current_ids=current_ids, data_inputs=data_inputs)
+        data_container = self._transform_core(data_container)
+
+        self.teardown()
```

Comment:
I would never call the teardown automatically at the end of a pipeline transform. 

I would instead do it on a [`__del__` method](https://stackoverflow.com/a/21514970/2476920) implemented in the BaseStep, although this would need to be tested, I'm not sure that recursive deletes would work.

Reply:
won't fix in this pr

Reply:
#55